src/git_clone_manager.rs

```diff
@@ -29,7 +29,12 @@ impl GitCloneManager {
         let session = self.github_app.new_session(owner, repo)?;
 
         let held_clone_dir = self.dir_pool.take_directory(session.github_host(), owner, repo);
-        self.clone_repo(&session, owner, repo, &held_clone_dir.dir())?;
+        if let Err(e) = self.clone_repo(&session, owner, repo, &held_clone_dir.dir()) {
```

Comment:
hmm. just now realized that `clone` is not the only place that uses the session token...  Maybe there is actually a more central place we should be doing this. :thinking: